core/src/avm1/globals/movie_clip_loader.rs

```diff
@@ -0,0 +1,283 @@
+            let listener = listeners
+                .get(&format!("{}", i), avm, context)?
+                .resolve(avm, context)?;
+
+            if let Value::Object(listener) = listener {
+                let handler = listener
+                    .get(&event_name, avm, context)?
+                    .resolve(avm, context)?;
+                handler
+                    .call(avm, context, listener, call_args)?
+                    .resolve(avm, context)?;
+            }
+        }
+    }
+
+    Ok(Value::Undefined.into())
+}
+
+pub fn load_clip<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let url = args
+        .get(0)
+        .cloned()
+        .unwrap_or(Value::Undefined)
+        .coerce_to_string(avm, context)?;
+    let target = args.get(1).cloned().unwrap_or(Value::Undefined);
+
+    if let Value::Object(target) = target {
+        if let Some(movieclip) = target
+            .as_display_object()
+            .and_then(|dobj| dobj.as_movie_clip())
+        {
+            let fetch = context.navigator.fetch(url, RequestOptions::get());
+            let process = context.load_manager.load_movie_into_clip(
+                context.player.clone().unwrap(),
+                DisplayObject::MovieClip(movieclip),
+                fetch,
+                Some(this),
+            );
+
+            context.navigator.spawn_future(process);
+        }
+
+        Ok(true.into())
+    } else {
+        Ok(false.into())
+    }
+}
+
+pub fn unload_clip<'gc>(
+    _avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let target = args.get(0).cloned().unwrap_or(Value::Undefined);
+
+    if let Value::Object(target) = target {
+        if let Some(mut movieclip) = target
+            .as_display_object()
+            .and_then(|dobj| dobj.as_movie_clip())
+        {
+            movieclip.unload(context);
+            movieclip.replace_with_movie(context.gc_context, None);
+
+            return Ok(true.into());
+        }
+    }
+
+    Ok(false.into())
+}
+
+pub fn get_progress<'gc>(
+    _avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let target = args.get(0).cloned().unwrap_or(Value::Undefined);
+
+    if let Value::Object(target) = target {
+        if let Some(movieclip) = target
+            .as_display_object()
+            .and_then(|dobj| dobj.as_movie_clip())
+        {
+            let ret_obj = ScriptObject::object(context.gc_context, None);
+            ret_obj.define_value(
+                context.gc_context,
+                "bytesLoaded",
+                movieclip
+                    .movie()
+                    .map(|mv| (mv.data().len() + 21).into())
```

Comment:
This should be the actual length of the compressed SWF file, but this is currently returning the uncompressed length. Also, the + 21 for the header isn't constant -- annoyingly the stage size rect in the header can be variable size (try making a 4000x4000 SWF). I know the size isn't easily available right now -- eventually we'll want to store the file length somewhere in `SwfMovie` or `SwfMovie.header`